### PR TITLE
Added Missing HTML5 elements

### PR DESCRIPTION
**Pull request by snebes**
Adding the following elements to the parser:

- article
- aside
- footer
- header
- main
- mark
- nav
- section
- time

Tests are included for these as well.

### CODE REVIEW

**Comment by tgalopin** [comment]
Hi @snebes !

Thanks for your PR!

However, I'm not sure why you would need to have "document" tags (article, aside, footer, header, main, nav, section) inside data sent by a user. What is your use-case here?

The others (time and mark) seems good ideas but I would not add them to the basic extension. Perhaps to the extra one?

File: src/Extension/Basic/NodeVisitor/TimeNodeVisitor.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+use HtmlSanitizer\Visitor\AbstractNodeVisitor;
+use HtmlSanitizer\Visitor\HasChildrenNodeVisitorTrait;
+use HtmlSanitizer\Visitor\NamedNodeVisitorInterface;
+
+/**
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @final
+ */
+class TimeNodeVisitor extends AbstractNodeVisitor implements NamedNodeVisitorInterface
```
Comment (tgalopin) [inline src/Extension/Basic/NodeVisitor/TimeNodeVisitor.php]: You probably want to allow the `datetime` attribute here :) .

Comment (snebes) [reply src/Extension/Basic/NodeVisitor/TimeNodeVisitor.php]: That's a good call.

**Comment by snebes** [comment]
@tgalopin I am using this library to validate HTML/XML from a variety of sources.  I only use some of these added elements in my version. Since these are all valid HTML5 structural markup, I felt that they should be a part of the core library.  Perhaps they should be grouped in the `Structure` namespace. (except `mark` and `time`)

**Comment by tgalopin** [comment]
This library was unfortunately not designed for the use-case of validating full HTML documents but instead to sanitize user input in order to display the result of rich editors safely. I would like to keep it precisely targeted on this to ease its maintenance.

I don't think integrating the "document" tags in the core library is a good fit with this aim. However, because of the very extensible nature of the library, you can definitely create your own extension to allow more tags in a customized way from outside of the library :) ! If you would like, you can even create a small open-source library to provide a "document" extension outside of the core.

Note that I would still be happy to merge this PR with only the `time` and `mark` elements in the extra extension!

Thanks!

**Comment by snebes** [comment]
@tgalopin I've dropped the unwanted elements from this PR.

**Comment by tgalopin** [comment]
Thanks! Can you also put the lasting tags in the extra extension?

**Comment by snebes** [comment]
@tgalopin reorganized as requested.

**Comment by tgalopin** [comment]
Great, thanks very much for your work and your dedication, really appreciated!

I merge this now, if you would like you can create another PR too for the documentation (in order to add these two elements in the list) :) ! If you prefer I can do it too.

**Pull request merged**